core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -765,7 +766,7 @@ public UserProfileService getUserProfileService() {
         if (unknownAttributes != null) {
             logger.warn("Attribute(s) {} not in the datafile.", unknownAttributes);
             // make a copy of the passed through attributes, then remove the unknown list
-            attributes = new HashMap<String, String>(attributes);
+            attributes = new HashMap<String, Object>(attributes);
```

Comment:
You should add a test that actually removes unknown attributes to see what happens here. But my understanding is that `Object` is actually different from `?`. 
See https://stackoverflow.com/questions/678822/what-is-the-difference-between-and-object-in-java-generics

Reply:
You can't use a question mark in a constructor.

Reply:
I actually think that the proper way to use the constructor is HashMap<>(attributes).  So, @wangjoshuah is correct.

Please update to something like:
HashMap<String, ?> attr = new HashMap<>();

Reply:
Looks like `<>` was not introduced until Java 7 and we're currently on Java 6. Thoughts? 

https://stackoverflow.com/questions/16612157/why-does-new-hashmap-produce-an-error-in-jdk-1-6-but-not-1-7?lq=1

Reply:
Okay, it sounds like
```
new HashMap<String, Object>();
```
is indeed the way to go...

Reply:
https://github.com/optimizely/java-sdk/pull/208 is the way to go. Good call @thomaszurkan-optimizely